### PR TITLE
[SPARK-33479][DOC] Make the API Key of DocSearch configurable

### PR DESCRIPTION
**Pull request by gengliangwang**
### What changes were proposed in this pull request?

Make the API key of DocSearch configurable and avoid hardcoding in the HTML template

### Why are the changes needed?

After https://github.com/apache/spark/pull/30292, our Spark documentation site supports searching.
However, the default API key always points to the latest release doc. We have to set different API keys for different releases. Otherwise, the search results are always based on the latest documentation(https://spark.apache.org/docs/latest/) even when visiting the documentation of previous releases.

As per discussion in https://github.com/apache/spark/pull/30292#issuecomment-725613417, we should make the API key configurable and set different values for different releases.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Manual test

### CODE REVIEW

File: docs/_config.yml [hunk 1]
```diff
@@ -26,3 +26,15 @@ SCALA_VERSION: "2.12.10"
 MESOS_VERSION: 1.0.0
 SPARK_ISSUE_TRACKER_URL: https://issues.apache.org/jira/browse/SPARK
 SPARK_GITHUB_URL: https://github.com/apache/spark
+# Before a new release, we should apply a new `apiKey` for the new Spark documentation
```
Comment (gengliangwang) [inline docs/_config.yml:29]: An alternative way is, we apply a new API key for the previous release instead. For example, 
we are going to release 3.1.1 and the current release is 3.1.0. We apply a new API key for https://spark.apache.org/docs/3.1.0 and publish it. After that, we publish 3.1.1 with the default API key which points to the latest doc.

**Comment by gengliangwang** [comment]
cc @beliefer @gatorsmile @maropu @sarutak , since you have reviewed https://github.com/apache/spark/pull/30292

File: docs/_config.yml [hunk 2]
```diff
@@ -26,3 +26,15 @@ SCALA_VERSION: "2.12.10"
...
+# on https://docsearch.algolia.com/. Otherwise, after release, the search results are always based
+# on the latest documentation(https://spark.apache.org/docs/latest/) even when visiting the
+# documentation of previous releases.
+DOCSEARCH_SCRIPT: |
+  docsearch({
+      apiKey: 'b18ca3732c502995563043aa17bc6ecb',
```
[the remark this answers is not in the document]
Comment (maropu) [reply docs/_config.yml:35]: It makes sense.

**Review by maropu — APPROVED** [review]

File: docs/_layouts/global.html [hunk 3]
```diff
@@ -187,15 +187,8 @@ <h1 class="title">{{ page.title }}</h1>
...
-            docsearch({
-                apiKey: 'b18ca3732c502995563043aa17bc6ecb',
-                indexName: 'apache_spark',
-                inputSelector: '#docsearch-input',
-                enhancedSearchInput: true,
-                debug: false // Set debug to true if you want to inspect the dropdown
-        });
+            {{site.DOCSEARCH_SCRIPT}}
         </script>
-
```
Comment (maropu) [inline docs/_layouts/global.html:198]: nit: an unnecessary change?

**Review by sarutak — APPROVED** [review]

**Comment by sarutak** [comment]
Seems fine to me except the minor issue @maropu pointed out.

By the way, though it's not the thing done in this PR, we should also update the [release process](http://spark.apache.org/release-process.html) right?

File: docs/_layouts/global.html (hunk 3 above)
Comment (gengliangwang) [reply docs/_layouts/global.html:198]: Thanks, updated.

**Comment by gengliangwang** [comment]
> By the way, though it's not the thing done in this PR, we should also update the release process right?

@sarutak sure, we should update it after this one is merged.

File: docs/_config.yml (hunk 1 above)
Comment (beliefer) [reply docs/_config.yml:29]: Yeah

**Review by beliefer — APPROVED** [review]
LGTM

**Comment by maropu** [comment]
Thanks! Merged to master.

**Comment by gengliangwang** [comment]
@maropu @sarutak @beliefer Thanks for the review.
As for updating the release process, I am trying to apply a new API key for https://spark.apache.org/docs/3.1.0 . I would like to see if I can get the API key before the doc site actually published.

**Comment by beliefer** [comment]
> @maropu @sarutak @beliefer Thanks for the review.
> As for updating the release process, I am trying to apply a new API key for https://spark.apache.org/docs/3.1.0 . I would like to see if I can get the API key before the doc site actually published.

Yes, we should try it.
